kernel/src/generic_main.rs

```diff
@@ -1,26 +1,16 @@
 use super::device_tree::DeviceTree;
-use super::driver_manager::DriverManager;
 use super::drivers::qemuexit::QemuExit;
-use super::drivers::{
-    pl011::Pl011,
-    // gicv2::GicV2
-    Console,
-    Driver,
-};
+use super::drivers::Driver;
 use super::globals;
-use core::sync::atomic::{AtomicUsize, Ordering};
 
 use crate::hal;
 use crate::mm::{alloc_pages_for_hal, map_address_space};
-use hal_core::mm::{PageMap, Permissions};
 
-use crate::executable::elf::Elf;
-use align_data::include_aligned;
-use align_data::Align4K;
+use crate::tests::{self, TestResult};
 
-use log::{info, trace};
+use log::info;
 
-pub fn generic_main(dt: DeviceTree, hacky_devices: &[&dyn Driver]) -> ! {
+pub fn generic_main<const LAUNCH_TESTS: bool>(dt: DeviceTree, hacky_devices: &[&dyn Driver]) -> ! {
```

Comment:
I mean why not but an argument is simpler

Reply:
That way I'm sure that stuff won't get compiled if we don't use it